### PR TITLE
mixed: apply fixes recommended by staticcheck and dead code removal

### PR DESCRIPTION
**Pull request by bhandras**
This PR fixes the most important recommendations from `staticcheck` while also removes some dead/unused code.

### CODE REVIEW

**Comment by bhandras** [comment]
Weirdly the unused `logClosure` isn't catched by `staticcheck` everywhere, but imho it could be removed from mostly everywhere. Also I can squash those commits to one of everyone agrees.

**Comment by joostjager** [comment]
Reviews are requested from too many people (five). Probably happened accidentally.

**Comment by bhandras** [comment]
> Reviews are requested from too many people (five). Probably happened accidentally.

I believe reviewers were auto selected based on code ownership. I'll remove a a few people.

File: lntest/harness.go [hunk 1]
```diff
@@ -211,12 +211,15 @@ func (n *NetworkHarness) SetUp(lndArgs []string) error {
 	// Now block until both wallets have fully synced up.
 	expectedBalance := int64(btcutil.SatoshiPerBitcoin * 10)
 	balReq := &lnrpc.WalletBalanceRequest{}
-	balanceTicker := time.Tick(time.Millisecond * 50)
+	balanceTicker := time.NewTicker(time.Millisecond * 50)
+	defer func() {
```
Comment (guggero) [inline lntest/harness.go:215]: Can't this just be `defer balanceTicker.Stop()`?

Comment (bhandras) [reply lntest/harness.go]: Of course, fixed.

**Comment by bhandras** [comment]
> Nice fixes, always love to see code quality improvements 100
> I can confirm the changes all look safe. Found just one small improvement.
> 
> What I asked myself was: If we already have `staticcheck` enabled in our linter (#4204), why didn't it find those fixes? Do we need to update `golangci-lint` or tune its settings?

Yeah weird. I'll investigate this.

**Comment by Roasbeef** [comment]
Landing this now so it doesn't have to go through rebase hell.

**Pull request merged**